chain/traversal.go

```diff
@@ -16,21 +16,19 @@ type BlockProvider interface {
 // GetParentTipSet returns the parent tipset of a tipset.
 // The result is empty if the tipset has no parents (including if it is empty itself)
 func GetParentTipSet(ctx context.Context, store BlockProvider, ts types.TipSet) (types.TipSet, error) {
-	newTipSet := types.TipSet{}
 	parents, err := ts.Parents()
-	if err != nil {
-		return nil, err
+	if err != nil || parents.Len() == 0 {
```

Comment:
breadcrumb: the second clause just short circuits the rest of the method. The results would be the same.

Reply:
Not quite. `NewTipSet` returns an error if given no blocks, but this returns Undef and no error (for the genesis tipset). I added a comment.